### PR TITLE
deploy: add support for dc --dry-run to rollout undo

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/11215

@kargakis PTAL

### CODE REVIEW

File: pkg/deploy/cmd/rollback.go [hunk 1]
```diff
@@ -49,9 +48,14 @@ func (r *DeploymentConfigRollbacker) Rollback(obj runtime.Object, updatedAnnotat
...
 	}
 
-	_, err = r.dn.DeploymentConfigs(config.Namespace).Update(rolledback)
-	if err != nil {
-		return "", err
+	if !dryRun {
+		if _, err = r.dn.DeploymentConfigs(config.Namespace).Update(rolledback); err != nil {
+			return "", err
+		}
+	} else {
```
Comment (kargakis) [inline pkg/deploy/cmd/rollback.go]: nit: reduce the diff by inverting this
```go
if dryRun {
  out := bytes.NewBuffer([]byte("\n"))
  kubectl.DescribePodTemplate(rolledback.Spec.Template, out)
  return out.String(), nil
}

_, err = r.dn.DeploymentConfigs(config.Namespace).Update(rolledback)
if err != nil {
  return "", err
}
```

**Comment by kargakis** [comment]
Test in test-cmd too

**Comment by mfojtik** [comment]
[test]

@kargakis PTAL

**Comment by kargakis** [comment]
LGTM - merge on green

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**